src/ITensorsExtensions/ITensorsExtensions.jl

```diff
@@ -57,7 +59,13 @@ function map_itensor(
 )
```

Comment:
It seems like this function should be called `map_singular_values`. Also I don't like using `first(inds(A))` as a default, generally ITensor functions should not depend on the ordering of the indices and that is a pretty arbitrary choice.

Reply:
Yeah `map_eigenvalues` (assuming we switch to `Eigen` based on below) seems reasonable. 

Agreed there is some arbitraryness in the choice of `linds` but tbh I don't really like the current default of looking for prime and unprimed pairs. I feel like if I send this function a tensor with `2` indices it should automatically work by treating it as a matrix which in the prime case it won't unless the tensor has indices `i` and `i`.

In our case we are most often going to be passing message tensors to this function and assuming that we construct the norm network `ψψ` by `simming` the link indices `ψ` then we presumably want a default that handles that.

Reply:
But it is also arbitrary to have special behavior for order 2 tensors, and in general we don't like to do that since it makes it more difficult to write generic ITensor code. It is safer to expect users to input the indices explicitly and not have a default, this is meant to be a general function and not be tailored to the use case of BP (where in general message tensors won't be order 2, in which case the default you are suggesting won't work and have surprising behavior).

Reply:
Yeah okay I can make it forward to `eigen` and inherit that behavior